org.envirocar.obd/src/main/java/org/envirocar/obd/adapter/CommandExecutor.java

```diff
@@ -107,6 +107,13 @@ public void call(Subscriber<? super byte[]> subscriber) {
             byteArray = baos.toByteArray();
         }
 
+        if (byteArray.length == 0){
+            LOGGER.info("Unexpected empty line anonmaly detected. Try to read next line.");
+            baos.reset();
+            readUntilLineEnd(baos);
```

Comment:
Can it happen that the thread gets blocked forever here? If there is really not data in it, the app is never be able to return to command sending mode.. just a thought.